[linux-6.6.y] Add support for Zhaoxin I2C controller

leoliu-oc opened the pull request:
zhaoxin inclusion
category: feature

-------------------

Zhaoxin I2C Linux driver support all bidirectional bus protocols speed specified in the I2C Specification 7.0.

opsiff commented on drivers/i2c/busses/Kconfig:
```diff
@@ -334,6 +334,18 @@ config I2C_VIAPRO
+config I2C_ZHAOXIN
+	tristate "Zhaoxin I2C controller driver"
+	depends on PCI
+	default m
```
为什么要default m,arm平台上也需要默认打开这个？

leoliu-oc replied:
这边会修改为依赖CONFIG_CPU_SUP_ZHAOXIN

opsiff reviewed (approved):
/ok-to-test
/lgtm

leoliu-oc committed:
Add support for Zhaoxin I2C controller

zhaoxin inclusion
category: feature

-------------------

Zhaoxin I2C Linux driver support all bidirectional bus protocols speed
specified in the I2C Specification 7.0.

Signed-off-by: leoliu-oc <leoliu-oc@zhaoxin.com>

opsiff commented:
/ok-to-test
